crates/sargon-uniffi/src/profile/mfa/security_structures/security_shield_builder.rs

```diff
@@ -378,6 +389,26 @@ impl SecurityShieldBuilder {
     }
 }
 
+use sargon::FactorSource as InternalFactorSource;
+
+#[uniffi::export]
+impl SecurityShieldBuilder {
+    pub fn auto_assign_factors_to_recovery_and_confirmation_based_on_primary(
+        &self,
+        all_factors: Vec<FactorSource>,
+    ) -> Result<()> {
+        let all_factors = all_factors
+            .into_iter()
+            .map(|x| x.into_internal())
+            .collect::<IndexSet<sargon::FactorSource>>();
```

Comment:
Let's define this in internal_mappings:
```rust
impl<InternalElement, Element> IntoInternal<Vec<Element>, IndexSet<InternalElement>>
    for Vec<Element>
where
    Element: Into<InternalElement>,
    InternalElement: std::hash::Hash + Eq,
{
    fn into_internal(self) -> IndexSet<InternalElement> {
        self.into_iter().map(Into::into).collect::<IndexSet<_>>()
    }
}
```

also the inverse conversion:
```rust
impl<InternalElement, Element> FromInternal<IndexSet<InternalElement>, Vec<Element>>
    for IndexSet<InternalElement>
where
    Element: From<InternalElement>,
{
    fn into_type(self) -> Vec<Element> {
        self.into_iter().map(Element::from).collect()
    }
}
```

Reply:
@GhenadieVP thx! I've added it here, **and made use of it in other files too**, very nice:

https://github.com/radixdlt/sargon/pull/299/commits/d4335a14a3e4f31781db1fe6f9f10e09178561dd

We should add one for `IndexMap` too

Reply:
it should be possible to achieve through some trait trickery, but it is a separate task on its own :), aka converting automatically from IndexMap<Key,Value> to some struct X { key, value }